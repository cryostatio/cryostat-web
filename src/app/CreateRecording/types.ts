/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { EventTemplate, KeyValue } from '@app/Shared/Services/api.types';
import { ValidatedOptions } from '@patternfly/react-core';

export type EventTemplateIdentifier = Pick<EventTemplate, 'name' | 'type'>;

interface _FormBaseData {
  name: string;
  template?: EventTemplateIdentifier;
  labels: KeyValue[];
  continuous: boolean;
  archiveOnStop: boolean;
  restart: boolean;
  duration: number;
  durationUnit: number;
  skipDurationCheck?: boolean;
  maxAge: number;
  maxAgeUnit: number;
  maxSize: number;
  maxSizeUnit: number;
  toDisk: boolean;
}

interface _FormValidationData {
  nameValid: ValidatedOptions;
  labelsValid: ValidatedOptions;
  durationValid: ValidatedOptions;
}

export type CustomRecordingFormData = _FormBaseData & _FormValidationData;
