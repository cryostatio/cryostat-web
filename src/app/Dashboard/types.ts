/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { FeatureLevel } from '@app/Shared/Services/service.types';
import { gridSpans, LabelProps } from '@patternfly/react-core';
import { Observable } from 'rxjs';

export interface Sized<T> {
  minimum: T;
  default: T;
  maximum: T;
}

export interface DashboardCardSizes {
  span: Sized<gridSpans>;
  height: Sized<number>;
}

export interface DashboardCardDescriptor {
  featureLevel: FeatureLevel;
  icon?: React.ReactNode;
  labels?: {
    content: string;
    color?: LabelProps['color'];
    icon?: React.ReactNode;
  }[];
  preview?: React.ReactNode;
  title: string;
  cardSizes: DashboardCardSizes;
  description: string;
  descriptionFull: JSX.Element | string;
  component: DashboardCardFC<DashboardCardTypeProps>;
  propControls: PropControl[];
  advancedConfig?: JSX.Element;
}
export type DashboardCardFC<P> = React.FC<P> & {
  cardComponentName: string;
};

export interface PropControlExtra {
  displayMapper?: (value: string) => string /* only has effect with 'select' PropControl kind */;
  min?: number;
  max?: number;
  [key: string]: unknown;
}

export interface PropControl {
  name: string;
  key: string;
  description: string;
  kind: 'boolean' | 'number' | 'string' | 'text' | 'select';
  values?: unknown[] | Observable<unknown>;
  defaultValue: unknown;
  extras?: PropControlExtra;
}

export interface DashboardCardTypeProps {
  span: number;
  dashboardId: number;
  isDraggable?: boolean;
  isResizable?: boolean;
  isFullHeight?: boolean;
  actions?: JSX.Element[];
}

export interface ValidationError {
  message: React.ReactNode;
}

export interface CardValidationResult {
  errors: ValidationError[];
  callForAction?: React.ReactNode;
}

export type LayoutTemplateFilter = 'Suggested' | 'Cryostat' | 'User-submitted';

export interface LayoutTemplateController {
  selectedTemplate: SelectedLayoutTemplate | undefined;
  setSelectedTemplate: (template: React.SetStateAction<SelectedLayoutTemplate | undefined>) => void;
  isUploadModalOpen: boolean;
  setIsUploadModalOpen: (isOpen: React.SetStateAction<boolean>) => void;
}

export interface CardConfig {
  id: string;
  name: string;
  span: gridSpans;
  props: object;
}

export type SerialCardConfig = Omit<CardConfig, 'id'>;

export const mockSerialCardConfig: SerialCardConfig = {
  name: 'Default',
  span: 12,
  props: {},
};
export interface DashboardLayout {
  name: string;
  cards: CardConfig[];
  favorite: boolean;
}

export type SerialDashboardLayout = Omit<DashboardLayout, 'name' | 'cards'> & { cards: SerialCardConfig[] };

// only name and vendor are needed to identify a template
export type LayoutTemplateRecord = Pick<LayoutTemplate, 'name' | 'vendor'>;

export enum LayoutTemplateVersion {
  'v2.3' = 'v2.3',
  'v2.4' = 'v2.4',
}

export enum LayoutTemplateVendor {
  BLANK = 'Blank',
  CRYOSTAT = 'Cryostat',
  USER = 'User-submitted',
}

export interface LayoutTemplate {
  name: string;
  description: string;
  cards: SerialCardConfig[];
  version: LayoutTemplateVersion;
  vendor: LayoutTemplateVendor;
}

export interface SelectedLayoutTemplate {
  template: LayoutTemplate;
  category: LayoutTemplateFilter;
}

export type SerialLayoutTemplate = Omit<LayoutTemplate, 'vendor'>;
