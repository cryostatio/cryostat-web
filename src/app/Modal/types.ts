/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export enum DeleteOrDisableWarningType {
  DeleteActiveRecordings = 'DeleteActiveRecordings',
  DeleteArchivedRecordings = 'DeleteArchivedRecordings',
  DeleteAutomatedRules = 'DeleteAutomatedRules',
  DisableAutomatedRules = 'DisableAutomatedRules',
  DeleteEventTemplates = 'DeleteEventTemplates',
  DeleteProbeTemplates = 'DeleteProbeTemplates',
  DeleteActiveProbes = 'DeleteActiveProbes',
  DeleteCredentials = 'DeleteCredentials',
  DeleteCustomTargets = 'DeleteCustomTargets',
  DeleteDashboardLayout = 'DeleteDashboardLayout',
  DeleteLayoutTemplate = 'DeleteLayoutTemplate',
  ClearDashboardLayout = 'ClearDashboardLayout',
}

export interface DeleteOrDisableWarning {
  id: DeleteOrDisableWarningType;
  title: string;
  label: string;
  description: string;
  ariaLabel: string;
}

export const DeleteActiveRecordings: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteActiveRecordings,
  title: 'Permanently delete your JFR recording?',
  label: 'Delete Active Recording',
  description: `If you click delete, recording and report data will be lost.`,
  ariaLabel: 'Recording delete warning',
};

export const DeleteArchivedRecordings: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteArchivedRecordings,
  title: 'Permanently delete your archived JFR recording?',
  label: 'Delete Archived Recording',
  description: `If you click delete, recording and report data will be lost.`,
  ariaLabel: 'Recording delete warning',
};

export const DeleteAutomatedRules: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteAutomatedRules,
  title: 'Permanently delete your Automated Rule?',
  label: 'Delete Automated Rule',
  description: `If you click delete, rule data will be lost.`,
  ariaLabel: 'Automated rule delete warning',
};

export const DisableAutomatedRules: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DisableAutomatedRules,
  title: 'Disable your Automated Rule?',
  label: 'Disable Automated Rule',
  description: `If you click Disable, the rule will be disabled.`,
  ariaLabel: 'Automated rule disable warning',
};

export const DeleteEventTemplates: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteEventTemplates,
  title: 'Permanently delete your Event Template?',
  label: 'Delete Event Template',
  description: `If you click delete, custom event template data will be lost.`,
  ariaLabel: 'Event template delete warning',
};

export const DeleteProbeTemplates: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteProbeTemplates,
  title: 'Permanently delete your Probe Template?',
  label: 'Delete Probe Template',
  description: `If you click delete, custom probe template data will be lost.`,
  ariaLabel: 'Probe template delete warning',
};

export const DeleteActiveProbes: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteActiveProbes,
  title: 'Permanently remove your Active Probes from the target?',
  label: 'Remove Active Probes',
  description: `If you click delete, active probes will be removed from the target.`,
  ariaLabel: 'Active Probes remove warning',
};

export const DeleteCredentials: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteCredentials,
  title: 'Permanently delete your Credentials?',
  label: 'Delete Credentials',
  description: `If you click delete, credential data for this target will be lost.`,
  ariaLabel: 'Credentials delete warning',
};

export const DeleteCustomTargets: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteCustomTargets,
  title: 'Permanently delete your Custom Target?',
  label: 'Delete Custom Targets',
  description: `If you click delete, custom target information will be lost.`,
  ariaLabel: 'Custom Targets delete warning',
};

export const DeleteDashboardLayout: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteDashboardLayout,
  title: 'Permanently delete your Dashboard Layout?',
  label: 'Delete Dashboard Layouts',
  description: `If you click delete, dashboard layout configuration data will be lost.`,
  ariaLabel: 'Dashboard Layout delete warning',
};

export const ClearDashboardLayout: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.ClearDashboardLayout,
  title: 'Permanently clear your Dashboard Layout?',
  label: 'Clear Dashboard Layout',
  description: 'If you click clear, all cards in the current dashboard layout will be removed.',
  ariaLabel: 'Dashboard Layout clear warning',
};

export const DeleteLayoutTemplate: DeleteOrDisableWarning = {
  id: DeleteOrDisableWarningType.DeleteLayoutTemplate,
  title: 'Permanently delete your Layout Template?',
  label: 'Delete Layout Template',
  description: `If you click delete, layout template configuration data will be lost.`,
  ariaLabel: 'Layout Template delete warning',
};

export const DeleteWarningKinds: DeleteOrDisableWarning[] = [
  DeleteActiveRecordings,
  DeleteArchivedRecordings,
  DeleteAutomatedRules,
  DisableAutomatedRules,
  DeleteEventTemplates,
  DeleteProbeTemplates,
  DeleteActiveProbes,
  DeleteCredentials,
  DeleteCustomTargets,
  DeleteDashboardLayout,
  DeleteLayoutTemplate,
  ClearDashboardLayout,
];
