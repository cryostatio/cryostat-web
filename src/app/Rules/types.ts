/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { EventTemplateIdentifier } from '@app/CreateRecording/types';
import { Rule } from '@app/Shared/Services/api.types';
import { ValidatedOptions } from '@patternfly/react-core';

export interface RuleToDeleteOrDisable {
  rule: Rule;
  type: 'DELETE' | 'DISABLE';
}

interface _FormBaseData {
  name: string;
  enabled: boolean;
  matchExpression: string;
  description: string;
  template?: EventTemplateIdentifier;
  maxAge: number;
  maxAgeUnit: number;
  maxSize: number;
  maxSizeUnit: number;
  archivalPeriod: number;
  archivalPeriodUnit: number;
  initialDelay: number;
  initialDelayUnit: number;
  preservedArchives: number;
}

interface _FormValidationData {
  nameValid: ValidatedOptions;
  matchExpressionValid: ValidatedOptions;
}

export type RuleFormData = _FormBaseData & _FormValidationData;

export const ruleObjKeys = [
  'name',
  'description',
  'matchExpression',
  'enabled',
  'eventSpecifier',
  'archivalPeriodSeconds',
  'initialDelaySeconds',
  'preservedArchives',
  'maxAgeSeconds',
  'maxSizeBytes',
];
