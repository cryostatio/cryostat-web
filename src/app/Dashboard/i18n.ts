/*
 * Copyright The Cryostat Authors
 *
 * The Universal Permissive License (UPL), Version 1.0
 *
 * Subject to the condition set forth below, permission is hereby granted to any
 * person obtaining a copy of this software, associated documentation and/or data
 * (collectively the "Software"), free of charge and under any and all copyright
 * rights in the Software, and any and all patent rights owned or freely
 * licensable by each licensor hereunder covering either (i) the unmodified
 * Software as contributed to or provided by such licensor, or (ii) the Larger
 * Works (as defined below), to deal in both
 *
 * (a) the Software, and
 * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
 * one is included with the Software (each a "Larger Work" to which the Software
 * is contributed by such licensors),
 *
 * without restriction, including without limitation the rights to copy, create
 * derivative works of, display, perform, and distribute the Software and make,
 * use, sell, offer for sale, import, export, have made, and have sold the
 * Software and the Larger Work(s), and to sublicense the foregoing rights on
 * either these or other terms.
 *
 * This license is subject to the following condition:
 * The above copyright notice and either this complete permission notice or at
 * a minimum a reference to the UPL must be included in all copies or
 * substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
/**
 * t('JvmDetailsCard.CARD_TITLE')
 * t('JvmDetailsCard.CARD_DESCRIPTION')
 * t('JvmDetailsCard.CARD_DESCRIPTION_FULL')
 * t('AutomatedAnalysisCard.CARD_TITLE')
 * t('AutomatedAnalysisCard.CARD_DESCRIPTION')
 * t('AutomatedAnalysisCard.CARD_DESCRIPTION_FULL')
 * t('CHART_CARD.JFR_METRICS_CARD_TITLE')
 * t('CHART_CARD.JFR_METRICS_CARD_DESCRIPTION')
 * t('CHART_CARD.JFR_METRICS_CARD_DESCRIPTION_FULL')
 * t('CHART_CARD.MBEAN_METRICS_CARD_TITLE')
 * t('CHART_CARD.MBEAN_METRICS_CARD_DESCRIPTION')
 * t('CHART_CARD.MBEAN_METRICS_CARD_DESCRIPTION_FULL')
 * t('CHART_CARD.PROP_CONTROLS.THEME.NAME')
 * t('CHART_CARD.PROP_CONTROLS.THEME.DESCRIPTION')
 * t('CHART_CARD.PROP_CONTROLS.THEME_COLOR.NAME')
 * t('CHART_CARD.PROP_CONTROLS.THEME_COLOR.DESCRIPTION')
 * t('CHART_CARD.PROP_CONTROLS.PERFORMANCE_METRIC.NAME')
 * t('CHART_CARD.PROP_CONTROLS.PERFORMANCE_METRIC.DESCRIPTION')
 * t('CHART_CARD.PROP_CONTROLS.DATA_WINDOW.NAME')
 * t('CHART_CARD.PROP_CONTROLS.DATA_WINDOW.DESCRIPTION')
 * t('CHART_CARD.PROP_CONTROLS.REFRESH_PERIOD.NAME')
 * t('CHART_CARD.PROP_CONTROLS.REFRESH_PERIOD.DESCRIPTION')
 * t('NonePlaceholderCard.CARD_TITLE')
 * t('NonePlaceholderCard.CARD_DESCRIPTION')
 * t('NonePlaceholderCard.CARD_DESCRIPTION_FULL')
 * t('AllPlaceholderCard.CARD_TITLE')
 * t('AllPlaceholderCard.CARD_DESCRIPTION')
 * t('AllPlaceholderCard.CARD_DESCRIPTION_FULL')
 **/
