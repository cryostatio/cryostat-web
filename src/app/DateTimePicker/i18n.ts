/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * t('TimeSpinner.INPUT_HOUR12_VALUE')
 * t('TimeSpinner.INPUT_HOUR24_VALUE')
 * t('TimeSpinner.INPUT_MINUTE_VALUE')
 * t('TimeSpinner.INPUT_SECOND_VALUE')
 * t('TimeSpinner.INCREMENT_HOUR12_VALUE')
 * t('TimeSpinner.INCREMENT_HOUR24_VALUE')
 * t('TimeSpinner.INCREMENT_MINUTE_VALUE')
 * t('TimeSpinner.INCREMENT_SECOND_VALUE')
 * t('TimeSpinner.DECREMENT_HOUR12_VALUE')
 * t('TimeSpinner.DECREMENT_HOUR24_VALUE')
 * t('TimeSpinner.DECREMENT_MINUTE_VALUE')
 * t('TimeSpinner.DECREMENT_SECOND_VALUE')
 */
