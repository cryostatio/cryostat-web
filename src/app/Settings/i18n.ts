/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * t('SETTINGS.AUTOMATED_ANALYSIS_CONFIG.TITLE')
 * t('SETTINGS.AUTOMATED_ANALYSIS_CONFIG.DESCRIPTION')
 * t('SETTINGS.CHARTS_CONFIG.TITLE')
 * t('SETTINGS.CHARTS_CONFIG.DESCRIPTION')
 * t('SETTINGS.AUTO_REFRESH.TITLE')
 * t('SETTINGS.AUTO_REFRESH.DESCRIPTION')
 * t('SETTINGS.CREDENTIALS_STORAGE.TITLE')
 * t('SETTINGS.CREDENTIALS_STORAGE.DESCRIPTION')
 * t('SETTINGS.CREDENTIALS_STORAGE.BROWSER_SESSION.TITLE')
 * t('SETTINGS.CREDENTIALS_STORAGE.BROWSER_SESSION.DESCRIPTION')
 * t('SETTINGS.CREDENTIALS_STORAGE.BACKEND.TITLE')
 * t('SETTINGS.CREDENTIALS_STORAGE.BACKEND.DESCRIPTION')
 * t('SETTINGS.DATETIME_CONTROL.TITLE')
 * t('SETTINGS.DATETIME_CONTROL.DESCRIPTION')
 * t('SETTINGS.DELETION_DIALOG_CONTROL.TITLE')
 * t('SETTINGS.DELETION_DIALOG_CONTROL.DESCRIPTION')
 * t('SETTINGS.FEATURE_LEVEL.TITLE')
 * t('SETTINGS.FEATURE_LEVEL.DESCRIPTION')
 * t('SETTINGS.LANGUAGE.TITLE')
 * t('SETTINGS.LANGUAGE.DESCRIPTION')
 * t('SETTINGS.NOTIFICATION_CONTROL.TITLE')
 * t('SETTINGS.NOTIFICATION_CONTROL.DESCRIPTION')
 * t('SETTINGS.THEME.TITLE')
 * t('SETTINGS.THEME.DESCRIPTION')
 * t('SETTINGS.WEBSOCKET_CONNECTION_DEBOUNCE.TITLE')
 * t('SETTINGS.WEBSOCKET_CONNECTION_DEBOUNCE.DESCRIPTION')
 * t('SETTINGS.CATEGORIES.CONNECTIVITY')
 * t('SETTINGS.CATEGORIES.GENERAL')
 * t('SETTINGS.CATEGORIES.NOTIFICATION_MESSAGE')
 * t('SETTINGS.CATEGORIES.DASHBOARD')
 * t('SETTINGS.CATEGORIES.ADVANCED')
 */
