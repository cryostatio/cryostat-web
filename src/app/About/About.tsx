/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import cryostatLogo from '@app/assets/cryostat_logo_hori_rgb_default.svg';
import cryostatLogoDark from '@app/assets/cryostat_logo_hori_rgb_reverse.svg';
import { BreadcrumbPage } from '@app/BreadcrumbPage/BreadcrumbPage';
import build from '@app/build.json';
import { ThemeSetting } from '@app/Settings/SettingsUtils';
import { useTheme } from '@app/utils/useTheme';
import { Brand, Card, CardBody, CardFooter, CardHeader } from '@patternfly/react-core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { AboutDescription } from './AboutDescription';

export interface AboutProps {}

export const About: React.FC<AboutProps> = (_) => {
  const { t } = useTranslation('public');
  const [theme] = useTheme();

  const logo = React.useMemo(() => (theme === ThemeSetting.DARK ? cryostatLogoDark : cryostatLogo), [theme]);

  return (
    <BreadcrumbPage pageTitle={t('About.ABOUT')}>
      <Card>
        <CardHeader>
          <Brand alt={build.productName} src={logo} className="cryostat-logo" />
        </CardHeader>
        <CardBody>
          <AboutDescription />
        </CardBody>
        <CardFooter>{t('CRYOSTAT_TRADEMARK', { ns: 'common' })}</CardFooter>
      </Card>
    </BreadcrumbPage>
  );
};

export default About;
