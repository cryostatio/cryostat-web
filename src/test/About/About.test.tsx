/*
 * Copyright The Cryostat Authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import i18n from '@app/../i18n/config';
import { About } from '@app/About/About';
import { ThemeSetting } from '@app/Settings/types';
import { defaultServices } from '@app/Shared/Services/Services';
import { cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import * as React from 'react';
import { I18nextProvider } from 'react-i18next';
import renderer, { act } from 'react-test-renderer';
import { of } from 'rxjs';
import { mockMediaQueryList, renderDefault, testT } from '../Common';
jest.mock('@app/BreadcrumbPage/BreadcrumbPage', () => {
  return {
    BreadcrumbPage: jest.fn((props) => {
      return (
        <div>
          {props.pageTitle}
          {props.children}
        </div>
      );
    }),
  };
});

jest.spyOn(defaultServices.settings, 'themeSetting').mockReturnValue(of(ThemeSetting.DARK));
jest.spyOn(defaultServices.settings, 'media').mockReturnValue(of(mockMediaQueryList));

jest.mock('@app/About/AboutDescription', () => {
  return {
    ...jest.requireActual('@app/About/AboutDescription'),
    AboutDescription: jest.fn(() => {
      return <div>AboutDescription</div>;
    }),
  };
});

describe('<About />', () => {
  afterEach(cleanup);

  it('renders correctly', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(<About />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('contains the correct information', async () => {
    renderDefault(
      <I18nextProvider i18n={i18n}>
        <About />
      </I18nextProvider>,
    );

    expect(screen.getByText('About')).toBeInTheDocument();
    const logo = screen.getByRole('img');
    expect(logo).toHaveClass('pf-c-brand cryostat-logo');
    expect(logo).toHaveAttribute('alt', 'Cryostat');
    expect(logo).toHaveAttribute('src', 'test-file-stub');
    expect(screen.getByText(testT('CRYOSTAT_TRADEMARK', { ns: 'common' }))).toBeInTheDocument();
  });
});
