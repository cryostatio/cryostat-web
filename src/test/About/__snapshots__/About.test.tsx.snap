// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> renders correctly 1`] = `
<div>
  About
  <div
    className="pf-v5-c-card"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF5/Card"
    data-ouia-safe={true}
    id=""
  >
    <div
      className="pf-v5-c-card__header"
    >
      <div
        className="pf-v5-c-card__header-main"
      >
        <img
          alt="Cryostat"
          className="pf-v5-c-brand cryostat-logo"
          src="test-file-stub"
          style={Object {}}
        />
      </div>
    </div>
    <div
      className="pf-v5-c-card__body"
    >
      <div>
        AboutDescription
      </div>
    </div>
    <div
      className="pf-v5-c-card__footer"
    >
      Copyright The Cryostat Authors, The Apache License, Version 2.0
    </div>
  </div>
</div>
`;
